### ISSUE TITLE
The signupform

### ISSUE THREAD

**Issue by tommygonzaleza** [breatheco-de/breatheco-de#7729]
The SignUpForm should be able to receive an array of strings or an array of objects where each object will have information about the form field. The object should look like this:

```js
{
    value: "",
    valid: false,
    required: true,
    type: "text",
    place_holder: "First name",
    error: "Please specify a valid first name",
}
```

It should be compatible with what was working before, the component should allow both methods of using it.

---

## Tests:

- [ ] Try to sign up from the landing of an interactive tutorial (Try it with and without adding a phone, it should let you sign up both ways)
- [ ] Try to sign up from the landing of a workshop (Try it with and without adding a phone, it should let you sign up both ways)
- [ ] Try to sign up from the landing of a project (Try it with and without adding a phone, it should let you sign up both ways)
- [ ] Try to sign up from the checkout process while getting a new plan (Try it with and without adding a phone, it should let you sign up only with the phone)

**Comment by tommygonzaleza** [comment]
- [ ] We should remove this message from the phone field
<img width="679" alt="Screenshot 2024-09-02 at 10 28 47 AM" src="https://github.com/user-attachments/assets/ff9c355b-0af2-4e2d-b4a7-e633f2b4df39">

**Comment by MadamSagas** [comment]
* I tried to sign up from the landing of an _interactive tutorial_. I was able to do it using the phone, but it wasn't possible to sign up without the phone number.

* I tried to sign up from the landing page of a workshop. I was able to do it using the phone, but it wasn't possible to sign up without it phone number input is required. Also, I found that the welcome video displayed after signing up is incorrect; instead, it’s a video of a soccer game. here's a screen record video 
https://github.com/user-attachments/assets/df3f0804-6edc-417a-93a8-9dd438e4fd62

I tried to sign up from the landing page of a project I was able to do it using the phone, but it wasn't possible to sign up without it phone number input is required. 


* I tried to sign up from the checkout process while getting a new plan. It does not let me sign up 
using only the phone number. Here's a screen record video 
https://github.com/user-attachments/assets/9708cdf3-02f3-489f-bf77-2b76bf2a72fd

Also the welcome video that is shown after sign up is incorrect 
![Image](https://github.com/user-attachments/assets/19b8afc1-e21b-4d70-918d-70d33fb4222b)
